stdlib/LinearAlgebra/src/dense.jl

```diff
@@ -563,7 +563,7 @@ function (^)(A::AbstractMatrix{T}, p::Real) where T
 
     # For integer powers, use power_by_squaring
     isinteger(p) && return integerpow(A, p)
-
+    isinteger(p*2) && return A^floor(Integer, p)*sqrt(A)
```

Comment:
Beware #55300.

Reply:
maybe check against `maxintfloat(x)` would help?

Reply:
we actually know that we won't be in this scenario since if we're above maxintfloat, `p` must be an Integer.

Reply:
if `p` is above `maxinfloat(p)`, then `floor(Integer,p)` fails with `InexactError`

Reply:
right, but if `p` is above `maxintfloat` than `isinteger(p)` is true, so we never reach this line.